# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: Apache-2.0

foundation/ace/ace_engine: disable graphics

The graphics rendering is slowing the system making some ACTS test suites to timeout.
Since we are aiming for certification for headless system, graphics stack 
should not be required.
Disabling graphics by skipping JSView::CreateComponent() execution.
In the future we will remove this patch when we the graphical part will be needed.

Apply to foundation/ace/ace_engine

Upstream-Status: Inappropriate

Signed-off-by: Francesco Pham <francesco.pham@huawei.com>
diff --git a/frameworks/bridge/declarative_frontend/jsview/js_view.cpp b/frameworks/bridge/declarative_frontend/jsview/js_view.cpp
index e455cb00..d58d5c79 100644
--- a/frameworks/bridge/declarative_frontend/jsview/js_view.cpp
+++ b/frameworks/bridge/declarative_frontend/jsview/js_view.cpp
@@ -254,6 +254,7 @@ RefPtr<OHOS::Ace::Component> JSView::CreateComponent()
 
     std::string key = ViewStackProcessor::GetInstance()->ProcessViewId(viewId_);
     auto composedComponent = AceType::MakeRefPtr<ComposedComponent>(key, "view");
+    return composedComponent;
 
     // add callback for element creation to component, and get pointer reference
     // to the element on creation. When state of this view changes, mark the
